packages/@atjson/document/src/index.ts

```diff
@@ -375,6 +375,63 @@ export default class Document {
     return matches;
   }
 
+  canonical(): Document {
+    let canonicalDoc = this.clone();
+    canonicalDoc.where({type: '-atjson-parse-token'}).update(a => {
+      canonicalDoc.deleteText(a.start, a.end);
+    });
+    canonicalDoc.where({type: '-atjson-parse-token'}).remove();
+    canonicalDoc.annotations = canonicalDoc.annotations.sort(this.sortByPositionAndTypeFunction);
+    return canonicalDoc;
+  }
+
+  sortByPositionAndTypeFunction(a: Annotation<any>, b: Annotation<any>): number {
+    let startDelta = a.start - b.start;
+    let endDelta = a.end - b.end;
+    if (startDelta === 0) {
+      if (endDelta === 0) {
+        return a.type.localeCompare(b.type);
```

Comment:
It might be worth having a check for type equality here; if two annotations have the same type, and are at the same position (but, e.g., have different attributes), the comparison may fail even if both documents are identical for our purposes except that the annotations got added in a different order.

I can imagine this being an extremely rare situation, so it's probably okay for now to just throw an exception or somehow make the error visible so that if it does happen we can see it and prioritize fixing it.

Reply:
@blaine
Hey! Should I just console.log, or is there a logger service that I couldnt find that we are using in @json

I dont think that we should throw though